src/controllers/appointment-controller.ts

```diff
@@ -0,0 +1,222 @@
+import express,{ Request, Response,NextFunction } from 'express';
+import { IAppointments } from '../models/appointment'; // Import the interface
+import AppointmentModel from '../models/appointment'; // Import the model
+import { IPatient } from '../models/patient'; // Import the interface
+import PatientModel from '../models/patient'; // Import the model
+
+//Definition of controller function for appointment
+
+export const getAllAppointmentForPatient = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const appointments = await AppointmentModel.find({ patientId: req.params.patId }).populate('patientId');
+        if (!appointments) {
```

Comment:
Need to introduce function. A function while will take any object and message and throw exception if object is null so we can remove if's from our code.

Reply:
Apply this technique in each function where you checking the null object